src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationService.kt

```diff
@@ -143,11 +143,12 @@ class NonAssociationsReconciliationService(
     val closedPlusOpenLists = nomisListSortedBySequence.partition { closedInNomis(it, today) }
     val nomisList = (closedPlusOpenLists.first + closedPlusOpenLists.second.takeLast(1))
       // needed to change sort order to date to compare against matching DPS records
-      .sortedBy { it.effectiveDate }
+      // when dates are the same then sort by typeSequence or id which should reflect the order they were created
+      .sortedWith(compareBy(NonAssociationResponse::effectiveDate, NonAssociationResponse::typeSequence))
 
     nomisTotalDetails += nomisList.size
 
-    val dpsList = dpsListUnsorted.sortedBy { it.whenCreated }
+    val dpsList = dpsListUnsorted.sortedWith(compareBy(NonAssociation::whenCreated, NonAssociation::id))
 
```

Comment:
I realised if i sort by date, _then by_ id or seq , that fixes the out-of-order discrepancy we have in the prod report atm